internal/graph/graph_test.go

```diff
@@ -39,19 +40,42 @@ var (
 )
 
 func TestRelationshipIngress_String(t *testing.T) {
-	require.Equal(t, "ingress type:\"document\" relation:\"viewer\", type ttu, tupleset type:\"document\" relation:\"parent\"", RelationshipIngress{
-		Type:             TupleToUsersetIngress,
-		Ingress:          typesystem.DirectRelationReference("document", "viewer"),
-		TuplesetRelation: typesystem.DirectRelationReference("document", "parent"),
-	}.String())
-	require.Equal(t, "ingress type:\"document\" relation:\"viewer\", type computed_userset", RelationshipIngress{
-		Type:    ComputedUsersetIngress,
-		Ingress: typesystem.DirectRelationReference("document", "viewer"),
-	}.String())
-	require.Equal(t, "ingress type:\"document\" relation:\"viewer\", type direct", RelationshipIngress{
-		Type:    DirectIngress,
-		Ingress: typesystem.DirectRelationReference("document", "viewer"),
-	}.String())
+	for _, tc := range []struct {
+		name     string
+		expected string
+		ingress  RelationshipIngress
+	}{
+		{
+			name:     "TupleToUsersetIngress",
+			expected: "ingress type:\"document\" relation:\"viewer\", type ttu, tupleset type:\"document\" relation:\"parent\"",
+			ingress: RelationshipIngress{
+				Type:             TupleToUsersetIngress,
+				Ingress:          typesystem.DirectRelationReference("document", "viewer"),
+				TuplesetRelation: typesystem.DirectRelationReference("document", "parent"),
+			},
+		},
+		{
+			name:     "ComputedUsersetIngress",
+			expected: "ingress type:\"document\" relation:\"viewer\", type computed_userset",
+			ingress: RelationshipIngress{
+				Type:    ComputedUsersetIngress,
+				Ingress: typesystem.DirectRelationReference("document", "viewer"),
+			},
+		},
+		{
+			name:     "DirectIngress",
+			expected: "ingress type:\"document\" relation:\"viewer\", type direct",
+			ingress: RelationshipIngress{
+				Type:    DirectIngress,
+				Ingress: typesystem.DirectRelationReference("document", "viewer"),
+			},
+		},
+	} {
+		t.Run(tc.name, func(t *testing.T) {
+			actual := strings.ReplaceAll(tc.ingress.String(), " ", "")
+			require.Equal(t, strings.ReplaceAll(tc.expected, " ", ""), actual)
```

Comment:
Feels a bit weird to to the trimming it the UT instead of in graph.go?  Is the failure pointing to inconsistency in the output message? Should we use strings.Trim(r.Ingress.String()) in graph.go?  (Though I am not sure if the same approach will fix for r.TuplesetRelation.String() as the whitespace is in the middle :( ).  We may need to do `strings.Join(strings.Fields(r.TuplesetRelation.String()), " ")`

Reply:
So the problem is in 

```go
func (x *RelationReference) String() string {
	return protoimpl.X.MessageStringOf(x)
}
```

For some reason that I don't know, this sometimes outputs one whitespace, and sometimes two. 

In the usage of this method, i don't really care which one it does, because it's used only in the UI of Jaeger.

Reply:
On second thought, i think what you said is a good idea. Updated the PR!